src/main/drivers/barometer/barometer_bmp280.c

```diff
@@ -57,91 +60,138 @@ STATIC_UNIT_TESTED bmp280_calib_param_t bmp280_cal;
 int32_t bmp280_up = 0;
 int32_t bmp280_ut = 0;
 
-static void bmp280_start_ut(void);
-static void bmp280_get_ut(void);
-#ifndef USE_BARO_SPI_BMP280
-static void bmp280_start_up(void);
-static void bmp280_get_up(void);
-#endif
+static void bmp280_start_ut(baroDev_t *baro);
+static void bmp280_get_ut(baroDev_t *baro);
+static void bmp280_start_up(baroDev_t *baro);
+static void bmp280_get_up(baroDev_t *baro);
+
 STATIC_UNIT_TESTED void bmp280_calculate(int32_t *pressure, int32_t *temperature);
 
+bool bmp280ReadRegister(busDevice_t *pBusdev, uint8_t reg, uint8_t length, uint8_t *data)
+{
+    switch (pBusdev->bustype) {
+#ifdef USE_BARO_SPI_BMP280
+    case BUSTYPE_SPI:
+        return spiReadRegisterBuffer(pBusdev, reg | 0x80, length, data);
+#endif
+#ifdef USE_BARO_BMP280
+    case BUSTYPE_I2C:
+        return i2cRead(pBusdev->busdev_u.i2c.device, pBusdev->busdev_u.i2c.address, reg, length, data);
```

Comment:
I would create `i2cReadRegisterBuffer` function analogous with `spiReadRegisterBuffer` and then use
```
return i2cReadRegisterBuffer(pBusdev, reg, length, data);
```